internal/dot/graph.go

```diff
@@ -105,6 +146,12 @@ type FailedNodes struct {
 	// TransitiveFailures is the list of nodes that failed to build due to
 	// missing/failed dependencies.
 	TransitiveFailures []*Result
+
+	// ctors is a collection of failed constructors IDs that are populated as the graph is traversed for errors.
```

Comment:
nit: let's try to wrap comments at 80 characters. There are no hard rules in Go about line length but for comments, we should aim for short and readable.